src/core/adapters/SimplePoolAdapter.ts

```diff
@@ -257,33 +268,44 @@ export abstract class SimplePoolAdapter implements IProtocolAdapter {
       eventResults.map(async (transferEvent) => {
         const {
           blockNumber,
-          args: { value },
+          args: { value: protocolTokenMovementValueRaw },
         } = transferEvent
 
         const protocolTokenPrice = await this.getPricePerShare({
           blockNumber,
           protocolTokenAddress,
         })
 
-        const pricePerShareRaw =
-          protocolTokenPrice.tokens?.[0]?.pricePerShareRaw
-        if (!pricePerShareRaw) {
-          throw new Error('No price for events at this time')
-        }
-
-        const movementValueRaw = BigInt(value.toString()) * pricePerShareRaw
         return {
           underlyingTokensMovement: underlyingTokens.reduce(
-            (accummulator, currentToken) => {
+            (accumulator, currentToken) => {
+              const currentTokenPrice = protocolTokenPrice.tokens?.find(
+                (price) => price.address === currentToken.address,
+              )
+
+              if (!currentTokenPrice) {
+                throw new Error('No price for underlying token at this time')
+              }
+
+              const protocolTokenMovementValue = formatUnits(
+                protocolTokenMovementValueRaw,
+                protocolTokenMetadata.decimals,
+              )
+
+              const movementValue =
+                +protocolTokenMovementValue * +currentTokenPrice.pricePerShare
```

Comment:
These are both decimal values. I think our formula and approach could do with a specification or mini white-paper to understand the implications fully, loss of precision is an acceptable risk, these are read-only values which will get converted into USD.